### PR TITLE
Beds: Check for buildable_to for bottom half

### LINKED ISSUE #991: Opening a door with a bed removes adjacent node
Using stock minetest and minetest_game.  @sofar @paramat 

screenshot_20160330_214242

https://www.youtube.com/watch?v=Cr8VfGkSQ6Y

### PR DESCRIPTION
**Pull request by sofar**
We properly checked top half already, just not the top half
target location.

Fixes #991

### CODE REVIEW

**Comment by Rui914** [comment]
An error occurs if there is unknown node.

File: mods/beds/api.lua [hunk 1]
```diff
@@ -59,6 +59,10 @@ function beds.register_bed(name, def)
 				return itemstack
 			end
 
+			if not minetest.registered_nodes[minetest.get_node(pos).name].buildable_to then
```
Comment (Rui914) [inline mods/beds/api.lua]: This should check existion definition.

**Comment by Rui914** [comment]
```
local def = minetest.registered_nodes[minetest.get_node(pos).name]
if not def or not def.buildable_to then
	return itemstack
end
```

**Comment by sofar** [comment]
I really hate doing all these fixes for unknown nodes... it is such a massive waste of time :(

**Comment by sofar** [comment]
Pushed additional check suggested by @Rui914 - as well as also doing that extra node nil check for the bed top position.

**Comment by Rui914** [comment]
>>I really hate doing all these fixes for unknown nodes...
I don't like that too.
Couldn't you find some way to improve it?

**Comment by paramat** [comment]
:+1:

**Comment by sofar** [comment]
@everamzah could you test and confirm this fixes? thanks!

**Comment by paramat** [comment]
+1 from sfan5 http://irc.minetest.ru/minetest-dev/2016-04-01#i_4571767

**Comment by paramat** [comment]
874b1ae326c146da1743c2975bddfea4767ef701

**Pull request closed without merging**